server.go

```diff
@@ -8,174 +8,203 @@ import (
 
 // Server is a go-socket.io server.
 type Server struct {
+	engine *engineio.Server
+
 	handlers map[string]*namespaceHandler
-	eio      *engineio.Server
 }
 
 // NewServer returns a server.
-func NewServer(c *engineio.Options) (*Server, error) {
-	eio, err := engineio.NewServer(c)
-	if err != nil {
-		return nil, err
-	}
+func NewServer(c *engineio.Options) *Server {
+	engine, _ := engineio.NewServer(c)
+
 	return &Server{
 		handlers: make(map[string]*namespaceHandler),
-		eio:      eio,
-	}, nil
+		engine:   engine,
+	}
 }
 
 // Close closes server.
 func (s *Server) Close() error {
-	return s.eio.Close()
+	return s.engine.Close()
 }
 
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	s.eio.ServeHTTP(w, r)
+	s.engine.ServeHTTP(w, r)
 }
 
 // OnConnect set a handler function f to handle open event for
 // namespace nsp.
 func (s *Server) OnConnect(nsp string, f func(Conn) error) {
-	h := s.getNamespace(nsp, true)
+	h := s.getNamespace(nsp)
+	if h == nil {
+		h = s.createNameSpace(nsp)
+	}
```

Comment:
Not sure if I think this is an improvement. Can you instead just add a second method: `getOrCreateNamespace` to reduce duplication and risk of bugs here?